src/core/plugin/luapi_application.h

```diff
@@ -1976,6 +1977,135 @@ static int applib_openFile(lua_State* L) {
     return 1;
 }
 
+/*
+ // * Adds images from the provided paths on the current page on the current layer.
+ // * Multiple images might be added with one call by passing multiple tables with parameters for images.
+ // *
+ // * Parameters per image:
+ // *  - path string: filepath to the image (required)
+ // *  - x number: x-Coordinate where to place the left upper corner of the image (default: 0)
+ // *  - y number: y-Coordinate where to place the left upper corner of the image (default: 0)
+ // *  scaling options:
+ // *  - width  integer: (default: -1)
+ // *  - height integer: (default: -1)
+ // *  - aspectRatio boolean: if only one of width and height is given should the image be scaled in the other dimension
+ // *    to preserve the aspect ratio? Is only set to false if the parameter is false, nil leads to the default value of
+ // *    true (default: true)
+ // *  - scale number: factor to apply to the both dimensions in the end after setting width/height (with/without
+ // *    keeping aspectratio) (default: 1)
+ // *
+ // * Returns nothing on success (lua will fill a variable set to the return value with nil) and an errorMsg on failure
+ // * (soft error as an error might be common for this function)
+ // *
+ // * Example: app.addImagesFromFilepath({path="/media/data/myImg.png", x=10, y=20, scale=2})
+ // * multiple images: app.addImagesFromFilepath({path="/media/data/myImg.png"}, {path="/media/data/myImg2.png"})
```

Comment:
Here you use a different format of passing multiple objects (compared with `addStrokes`, `addSplines`). It is convenient, but has the downside that you can't add any parameters (like `allowUndoRedoAction`) that concern all images at once. I would suggest that multiple objects are put into a table like in `addStrokes`.